src/fragments/lib/storage/flutter/configureaccess/50_customization.mdx

```diff
@@ -0,0 +1,86 @@
+## Customization
+
+### Customize Object Key Path
+
+As mentioned above, configuring an access level in storage upload options will cause your files to be uploaded with a _public/_ _protected/_ or _private/_ prefix.
+
+If you would like your items to be stored with a different prefix, you can customize your key path by defining a prefix resolver:
+
+```dart
+// Define your own prefix resolver, that conforms to `StorageS3PrefixResolver`
+class MyCustomPrefixResolver implements StorageS3PrefixResolver {
+  @override
+  Future<String> resolvePrefix({
+    required StorageAccessLevel storageAccessLevel,
+    String? identityId,
+  }) async {
+    switch (storageAccessLevel) {
+      case StorageAccessLevel.guest:
+        return 'myPublicPrefix/';
+      case StorageAccessLevel.protected:
+        return 'myProtectedPrefix/${identityId!}/';
+      case StorageAccessLevel.private:
+        return 'myPrivatePrefix/${identityId!}/';
+    }
+  }
+}
+```
+
+Then configure the storage plugin with the resolver.
+
+```dart
+final storagePlugin = AmplifyStorageS3(prefixResolver: MyCustomPrefixResolver());
+await Amplify.addPlugin(storagePlugin);
+```
+
+You will need to update your permissions to allow access to your custom prefix. Go to s3.console.aws.amazon.com then go to IAM -> Roles.
```

Comment:
I was reading up about this - I think if you make the change just from the AWS Console and not include that somehow in youyr project configuration for Amplify - the next time you do an Amplify push these policies might be removed. I think there has to be another way to do this either by using overrides or extensibility. Can we ask the CLI team about this please?

Reply:
@AaronZyLee would you know if that happens?